test/modules/TestAnalyzer/TestResultReporter.cs

```diff
@@ -101,16 +101,46 @@ static ModuleMessagesReport GetReceivedMessagesReport(string moduleId, double to
                 }
             }
 
-            // check if last message is older
-            if (DateTime.Compare(lastMessageDateTime.AddMilliseconds(toleranceInMilliseconds), endDateTime) < 0)
+            bool areMessagesMissing = missingCounter > 0;
```

Comment:
As talked, suggest to move these new logic into a function and returns status code and status message.